File: omni-backend/omni-crm/src/main/java/com/omni/crm/entity/CrmContact.java
package com.omni.crm.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;

/** CRM 客户联系人实体。 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("crm_contact")
public class CrmContact extends CrmOwnedEntity {
    @Serial private static final long serialVersionUID = 1L;
    /** 客户 ID */ private Long customerId;
    /** 姓名 */ private String name;
    /** 部门 */ private String department;
    /** 职位 */ private String jobTitle;
    /** 手机 */ private String mobile;
    /** 电话 */ private String phone;
    /** 邮箱 */ private String email;
    /** 决策角色 */ private String decisionRole;
    /** 主要联系人标记 */ private Integer primaryFlag;
    /** 状态 */ private Integer status;
}

File: omni-frontend/src/api/crm-contact.ts
/** @module api/crm-contact CRM 联系人管理 API。 */
import request from './request'
import type { ApiResponse, PageResult } from '@/types/api'

export interface CrmContact {
  id: number
  customerId: number
  name: string
  department: string | null
  jobTitle: string | null
  mobile: string | null
  phone: string | null
  email: string | null
  decisionRole: string | null
  primaryFlag: number
  status: number
  ownerUserId: number
  ownerUnitId: number | null
  ownerName: string | null
  ownerUnitName: string | null
  version: number
  createTime: string
  updateTime: string
}

export interface ContactQuery {
  keyword?: string
  customerId?: number
  status?: number
  page: number
  size: number
}

export interface CreateContactRequest {
  name: string
  department?: string
  jobTitle?: string
  mobile?: string
  phone?: string
  email?: string
  decisionRole?: string
  primary?: boolean
}

export interface UpdateContactRequest extends Omit<CreateContactRequest, 'primary'> {
  status?: number
  version: number
}

export function listContacts(params: ContactQuery) {
  return request.get<ApiResponse<PageResult<CrmContact>>>('/crm/contact/list', { params })
}

export function listCustomerContacts(customerId: number, params: { page: number; size: number }) {
  return request.get<ApiResponse<PageResult<CrmContact>>>(`/crm/customer/${customerId}/contact/list`, { params })
}

export function getContact(id: number) {
  return request.get<ApiResponse<CrmContact>>(`/crm/contact/${id}`)
}

export function createContact(customerId: number, data: CreateContactRequest) {
  return request.post<ApiResponse<CrmContact>>(`/crm/customer/${customerId}/contact`, data)
}

export function updateContact(id: number, data: UpdateContactRequest) {
  return request.put<ApiResponse<CrmContact>>(`/crm/contact/${id}`, data)
}

export function deleteContact(id: number, version: number) {
  return request.delete<ApiResponse<void>>(`/crm/contact/${id}`, { params: { version } })
}

export function setPrimaryContact(id: number, version: number) {
  return request.post<ApiResponse<CrmContact>>(`/crm/contact/${id}/primary`, { version })
}
